common/environment/environment.cpp

```diff
@@ -1821,9 +1791,45 @@ CConstDropZoneServerInfoIterator::CConstDropZoneServerInfoIterator(const IConstD
 {
     Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
     constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
-    Owned<IPropertyTree> pSrc = &dropZone->getPTree();
-    serverListIt.setown(pSrc->getElements("ServerList"));
-    maxIndex = pSrc->getCount("ServerList");
+
+    // For backward compatibility
+    SCMStringBuffer dropZoneMachineName;
+    // Returns dropzone '@computer' value if it is exists
+    dropZone->getComputerName(dropZoneMachineName);
+
+    if (0 != dropZoneMachineName.length())
+    {
+        // Create a ServerList for legacy element.
+        legacyServerList.setown(createPTree());
```

Comment:
See previous comment, better to declare legacyServerList here, getElements will link it, so retrained/valid because serverListIt is kept.